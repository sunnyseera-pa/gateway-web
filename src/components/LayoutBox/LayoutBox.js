/** @jsx jsx */
import { cx } from '@emotion/css';
import { jsx } from '@emotion/react';
import { addCommonPropTypes } from '../../configs/propTypes';
import useCommonStyles from '../../hooks/useCommonStyles';

const LayoutBox = ({ children, mt, mb, ml, mr, width, minWidth, maxWidth }) => {
	const commonStyles = useCommonStyles({ mt, mb, ml, mr, width, minWidth, maxWidth });

	return <div className={cx('ui-LayoutBox', commonStyles)}>{children}</div>;
};

LayoutBox.propTypes = addCommonPropTypes({});

export default LayoutBox;
