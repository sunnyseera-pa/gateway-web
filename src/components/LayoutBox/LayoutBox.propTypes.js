import PropTypes from 'prop-types';
import { addCommonPropTypes } from '../../configs/propTypes';

export const PROP_TYPES_LAYOUTBOX = addCommonPropTypes({
    display: PropTypes.string,
    alignItems: PropTypes.string,
    justifyContent: PropTypes.string,
    p: PropTypes.number,
    pl: PropTypes.number,
    pb: PropTypes.number,
    pt: PropTypes.number,
    pr: PropTypes.number,
    flexGrow: PropTypes.string,
});
