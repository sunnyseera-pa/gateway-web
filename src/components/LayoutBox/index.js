import LayoutBox from './LayoutBox';

export default LayoutBox;
