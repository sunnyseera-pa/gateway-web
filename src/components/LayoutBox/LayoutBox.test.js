import React from 'react';
import { render } from '@testing-library/react';
import LayoutBox from '.';

let wrapper;

const props = {
    children: 'Component goes here',
};

describe('Given the LayoutBox component', () => {
    describe('When it is rendered', () => {
        beforeAll(() => {
            wrapper = render(<LayoutBox {...props} />, { wrapper: Providers });
        });

        it('Then matches the previous snapshot', () => {
            expect(wrapper.container).toMatchSnapshot();
        });

        it('Then has the correct text', () => {
            expect(wrapper.getByText('Component goes here')).toBeTruthy();
        });
    });
});
