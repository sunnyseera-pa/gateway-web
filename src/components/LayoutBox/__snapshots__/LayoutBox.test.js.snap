// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the LayoutBox component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="ui-LayoutBox css-1sivg29"
  >
    Component goes here
  </div>
</div>
`;
